assets/src/blocks/Media/MediaEditor.js

```diff
@@ -17,10 +17,24 @@ const MediaInspectorOptions = ({ attributes, setAttributes }) => {
   // https://github.com/facebook/react/issues/14904
   // https://github.com/facebook/react/issues/955#issuecomment-469352730
   const [ media_url, setMediaURL ] = useState(attributes.media_url);
-  const debouncedMediaURLUpdate = useCallback(debounce(value => setAttributes({ media_url: value }), 300), []);
+  const debouncedMediaURLUpdate = useCallback(debounce(value => {
+    setAttributes({ embed_html: null });
 
-  function onSelectImage({id}) {
-    setAttributes({video_poster_img: id});
+    const embedPreview = wp.data.select('core').getEmbedPreview(resolveURL(value));
+
+    const embed_html = embedPreview ? embedPreview.html : null;
+
+    setMediaURL(value);
```

Comment:
As discussed by chat, this is already set in the onChange listener, it has to be there so the text is immediately updated in the input, otherwise there'll be a 300ms delay while typing.

Reply:
We ended up using the `defaultValue` value prop instead to resolve the issue